content/general/concepts/cicd-pipeline/cicd-pipeline.md

```diff
@@ -0,0 +1,29 @@
+---
+Title: 'CICD Pipeline'
+Description: 'The CICD pipeline automates the software/application delivery process.'
+Subjects: 
+  - 'Developer Tools'
+  - 'Mobile Development'
+  - 'Web Design'
+  - 'Web Development'
+Tags:
+  - 'Developer Tools'
+  - 'Development'
+CatalogContent:
+  - 'paths/front-end-engineer-career-path'
+  - 'paths/back-end-engineer-career-path'
+---
+
+The **CICD Pipeline** is the combination of continuous integration, delivery, and deployment of software development and application deployment.
+
+CI (Continuous Integration) is the automation process for Developers. CD (Continuous Delivery and Continuous Deployment) is the automated delivery of code to the testing and development steps of the pipeline. These environments are usually in a repository like Github. Continuous deployment refers to code passing automated tests and then placed in production, which in turn is used for automatic deployment.
+
+## Four Steps in CICD Pipeline
+
+1. Source: The creation of code by developers, which is continuously integrated within a shared repository. Version control systems, like Git are used for tracking and managing changes to the software. When changes are commited to the repository, automated builds and tests are triggered.
+
+2. Build: The process of using code from the repository to link necessary dependencies and libraries to "build" an executable file. Scripts can be utilized to convert the file in to a Virtual Machine or Container. Logs generated during this time are used to find errors and send notifications to developers when the build is complete. 
```

Comment:
```suggestion
2. Build: The process of using code from the repository to link necessary dependencies and libraries to "build" an executable file. Scripts can be utilized to convert the file to a [virtual machine](https://www.codecademy.com/resources/docs/general/virtual-machines) or container. Logs generated during this time are used to find errors and send notifications to developers when the build is complete. 
```

Reply:
Thank you for helping me make this a better document and finding the errors that I missed! All should be updated now!